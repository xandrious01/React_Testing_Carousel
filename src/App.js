import "./App.css";
import Carousel from "./Carousel";
import photos from "./photos.js";

function App() {
  
  const carouselPhotos = photos;
  const carouselTitle = "Shells from far-away beaches";

  return (
    <div className="App">
      <Carousel photos={carouselPhotos} title={carouselTitle} />
    </div>
  );
}

export default App;
