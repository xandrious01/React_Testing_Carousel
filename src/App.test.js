import { render, fireEvent } from "@testing-library/react";
import App from "./App";
import TEST_IMAGES from "./_testCommon.js";

it("renders without crashing", () => {
  <App />
});

it("matches snapshot", () => {
    const {asFragment} = render(<App/>);
    expect(asFragment()).toMatchSnapshot();
});