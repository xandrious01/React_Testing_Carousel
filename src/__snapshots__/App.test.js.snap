// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Carousel"
    >
      <h1>
        Shells from far-away beaches
      </h1>
      <div
        class="Carousel-main"
      >
        <i
          class="bi bi-arrow-left-circle"
          hidden=""
        />
        <div
          class="Card"
        >
          <h4
            class="Card-title"
          >
            Photo by Richard Pasquarella on Unsplash
          </h4>
          <img
            alt="Photo by Richard Pasquarella on Unsplash"
            class="Card-image"
            src="image1.jpg"
          />
          <small
            class="Card-small"
          >
            Image 1 of 3.
          </small>
        </div>
        <i
          class="bi bi-arrow-right-circle"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
